### PR TITLE
Support ipFamilyPolicy on services

### PR DESCRIPTION
**Pull request by UXabre**
Signed-off-by: Arend Lapere <arend.lapere@gmail.com>

### Description
Add the ability to set a service as being a dual stack service for both dashboard as cluster.
 
### Issues Resolved

- https://github.com/opensearch-project/helm-charts/issues/374

 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by jordarlu** [comment]
Thanks for the PR, we will review it and get back to you. Thanks!!

File: charts/opensearch-dashboards/templates/service.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if (semverCompare ">= 1.23-0" .Capabilities.KubeVersion.Version) }}
+  ipFamilyPolicy: {{ .Values.service.ipFamilyPolicy }}
```
Comment (prudhvigodithi) [inline charts/opensearch-dashboards/templates/service.yaml:17]: Hey @UXabre thanks for PR, this is good change to have, can you please let us know what would be the default value, is this a breaking change if a chart is updated with `helm upgrade` ?
Thank you

**Review by TheAlgo — COMMENTED** [review]
Agree with @prudhvigodithi, lets check on backward compatibility

File: charts/opensearch-dashboards/templates/service.yaml (hunk 1 above)
Comment (UXabre) [reply charts/opensearch-dashboards/templates/service.yaml:17]: Hi, I actually set the default value to SingleStack to be more compatible (as many people don't use dual stack yet). However, now I'm thinking about it, I think it's safer to just check if the variable is set and thus fallback to the default that the CNI provides (at least, to my understanding the CNI would define the implicit default).

Also, another change, to be more versatile would be to also provide a config for the IPFamilies as well, this way, one could set it to singlestack and ipv6 (if they are brave :p).

Curious to hear what you think of this proposal.

**Comment by UXabre** [comment]
Hi, I've added a check to see if ipFamilyPolicy is set, not setting it reverts to following the CNI defaults instead (so is the most compatible way imaginable). I've also included a commit with the ipFamilies field, this gives the most versatility to this PR. If you want, I can squash this into one commit to not pollute the git log.

**Review by TheAlgo — COMMENTED** [review]
Could you please add some documentation for this newly added `ipFamilyPolicy`

**Comment by prudhvigodithi** [comment]
Hey @UXabre thanks for the contribution, documentation is the [readme](https://github.com/opensearch-project/helm-charts/blob/main/charts/opensearch/README.md) file, can you please update this.
Thank you

**Comment by peterzhuamazon** [comment]
Hi @prudhvigodithi seems like @UXabre has updated the PR.
Let us review this again now?

Thanks.

**Comment by UXabre** [comment]
Hi, I've added the options to the readme as well as a link to the official k8s documentation regarding this. 
Also had to resolve some merge conflicts :-)

**Review by TheAlgo — APPROVED** [review]
Thanks a lot @UXabre

**Comment by TheAlgo** [comment]
@prudhvigodithi Lets get this to closure whenever you are free. Thanks

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @UXabre merged this PR, can you please backport this change to 1.x branch?
Thank you

**Comment by UXabre** [comment]
OK, will make a PR soon for the backport

**Comment by peterzhuamazon** [comment]
> OK, will make a PR soon for the backport

Hi @UXabre any progress on the backport?
Thanks.

**Comment by peterzhuamazon** [comment]
I backport it:
* https://github.com/opensearch-project/helm-charts/pull/412

**Pull request merged**